docs/concepts/architecture/cloud-controller.md

```diff
@@ -0,0 +1,249 @@
+
+#### Service Controller
+
+The Service Controller is responsible for listening to service create, update, and delete events. Based on the current state of the services in Kubernetes, it configures cloud load balancers (Such as ELB, or Google LB) to reflect the state of the services in Kubernetes.
+
+#### Persistent Volume Labels Controller
+
+The Persistent Volume Labels Controller applies labels on AWS EBS, and GCE PD volumes when they are created. This removes the need for users to manually set the labels on these volumes. 
+
+These labels are essential for the scheduling of pods, as these volumes are constrained to work only within the region/zone that they are in, and therefore any pod using these volumes need to be scheduled in the same region/zone.
+
+### 2. Kubelet
+
+The Node Controller contains the cloud dependent functionality of Kubelet.
+
+### 3. Kube APIServer
+
+The Persistent Volume Labels controller moves the cloud dependent functionality of Kube APIServer to CCM. 
+
+Plugin Mechanism
+------------------
+
+The Cloud Controller Manager uses Go interfaces to allow implementations from any cloud to be plugged in. Specifically, it uses the CloudProvider Interface defined [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/cloud.go)
+
+Any vendor that wants to plugin their cloud into Kubernetes only need to satisfy that interface. The implementation of the controllers and the scaffolding will be in core, but the implementation will exec out to the cloud interfaces, so long as the interface is satisfied. 
+
+To find more information about developing plugins, refer [here](https://kubernetes.io/docs/tasks/administer-cluster/developing-cloud-controller-manager/)
+
+Authorization
+--------------
+
+This section breaks down the access required on various API objects by CCM to perform its operations. 
+
+#### Node Controller
+
+The Node Controller only works with node objects. It requires full access to get, list, create, update, patch, watch, and delete node objects.
+
+v1/Node: 
+- Get
+- List
+- Create
+- Update
+- Patch
+- Watch
+
+#### Route Controller
+
+The Route Controller listens to node object creation and configures routes appropriately. It requires get access to node objects. 
+
+v1/Node: 
+- List
+
+#### Service Controller
+
+The Service Controller listens to service create, update and delete events and then configures endpoints for those services appropriately. 
+
+In order to access services it requires list, and watch access. In order to update services, it requires patch and update access. 
+
+In order to setup endpoints for the services, it requires access to create, list, get, watch, and update.
+
+v1/Service:
+- List
+- Watch
+- Patch
+- Update
+
+v1/Endpoint:
+- Create
+- List
+- Get
+- Watch
+- Update
```

Comment:
The service controller also watches node objects for backend updates. 

IIRC, this controller doesn't have any dependencies on Endpoints.

Reply:
@jhorwit2 I did my own research too. It looks like giving access to endpoint is a security loophole. Good find. I'll update it.